File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int length = nums.length;
        List<Integer> data = new ArrayList<>();
        for(int i = 0;i<length;i++){
            if(nums[i]!=val){
                //System.out.println(nums[i]);
                data.add(nums[i]);
            }
        }

 
        for(int i = 0;i<data.size();i++){
            nums[i] = data.get(i);
           
        }
        
        return data.size();
    }
}

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int length = nums.size();
        vector<int> indexList;
        for(int i = 0;i<length;i++){
            if(nums[i] == val){
                indexList.emplace_back(i);
                nums.emplace_back(nums[i]);
            }
        }
        
        for(int i = 0;i<indexList.size();i++){
            indexList[i] -= i;
        }
        
        for(int n: indexList){
            nums.erase(nums.begin() + n);
        }
        
        return nums.size()-indexList.size();
    }
};